File: template/{{ cookiecutter._template }}/{{ cookiecutter.project_name|lower|replace(' ', '_') }}_android/android/src/main/kotlin/za/co/discovery/flutter/{{ cookiecutter.project_name|lower|replace(' ', '_') }}/method_handler/MethodResultWrapper.kt
package za.co.discovery.flutter.{{ cookiecutter.project_name|lower|replace(' ', '_') }}.method_handler

import io.flutter.plugin.common.MethodChannel

class MethodResultWrapper :
    MethodChannel.Result {
    lateinit var methodResult: MethodChannel.Result
    fun setResult(methodResult: MethodChannel.Result) {
        this.methodResult = methodResult
    }

    override fun success(result: Any?) {
        methodResult.success(result)
    }

    override fun error(errorCode: String, errorMessage: String?, errorDetails: Any?) {
        methodResult.error(
            errorCode,
            errorMessage,
            errorDetails
        )
    }

    override fun notImplemented() {
        methodResult.notImplemented()
    }
}

File: template/{{ cookiecutter._template }}/{{ cookiecutter.project_name|lower|replace(' ', '_') }}_ios/ios/Classes/method_handler/MethodResultWrapper.swift
import Flutter

class MethodResultWrapper: NSObject {
    private var methodResult: FlutterResult!
    
    func setResult(methodResult: @escaping FlutterResult) {
        self.methodResult = methodResult
    }
    
    func success(_ result: Any?) {
        DispatchQueue.main.async {
            self.methodResult(result)
        }
    }
    
    func error(_ errorCode: String, errorMessage: String?, errorDetails: Any?) {
        DispatchQueue.main.async {
            self.methodResult(FlutterError(code: errorCode, message: errorMessage, details: errorDetails))
        }
    }
    
    func notImplemented() {
        DispatchQueue.main.async {
            self.methodResult(FlutterMethodNotImplemented)
        }
    }
}
